src/app/3d/qgs3dmapcanvas.cpp

```diff
@@ -108,3 +109,53 @@ void Qgs3DMapCanvas::saveAsImage( const QString fileName, const QString fileForm
     mEngine->requestCaptureImage();
   }
 }
+
+void Qgs3DMapCanvas::setMapTool( Qgs3DMapTool *tool )
+{
+  if ( tool == mMapTool )
+    return;
+
+  if ( mMapTool && !tool )
+  {
+    mEngine->window()->removeEventFilter( this );
+    mScene->cameraController()->setEnabled( true );
+    mEngine->window()->setCursor( Qt::OpenHandCursor );
+  }
+  else if ( !mMapTool && tool )
+  {
+    mEngine->window()->installEventFilter( this );
+    mScene->cameraController()->setEnabled( false );
+    mEngine->window()->setCursor( Qt::CrossCursor );
```

Comment:
Should this call a (virtual) method on the tool to determine the cursor?

Reply:
At some point yes when we need different cursors for different tools...

Reply:
It just looks like code that belongs into a specific map tool (which already exists) and not into the canvas...

Reply:
cf87b85